### ISSUE TITLE
lgb.cv for regression results in fold error (R)

### ISSUE THREAD

**Issue by GlobalMin** [Microsoft/LightGBM#896]
Regression with lgb.cv() doesn't seem to work. When I run the basic example with a binary outcome it works but for regression the target variable is usually continuous. If I change the example to have a continuous target the error appears.

## Environment info
Operating System: Windows
CPU: x86_64-w64-mingw32/x64 (64-bit)
R version: 3.4.0 (2017-04-21)

## Error Message:
In foldVector[y == dimnames(numInClass)$y[i]] <- sample(seqVector) :
  number of items to replace is not a multiple of replacement length

## Reproducible examples
data(agaricus.train, package='lightgbm')
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label=runif(6513,0,100))
params <- list(objective="regression", metric="l2")
model <- lgb.cv(params, dtrain, 10, nfold=5, min_data=1, learning_rate=1, early_stopping_rounds=10)

**Comment by Laurae2** [comment]
@GlobalMin Can you try https://github.com/Microsoft/LightGBM/pull/897 ?

**Comment by GlobalMin** [comment]
Yes that works perfectly thank you!
